packages/fxa-settings/src/pages/PostVerify/ThirdPartyAuthCallback/index.tsx

```diff
@@ -140,6 +104,54 @@ const ThirdPartyAuthCallback = ({
     [finishOAuthFlowHandler, integration, location.search]
   );
 
+  const verifyThirdPartyAuthResponse = useCallback(async () => {
+    const { code: thirdPartyOAuthCode, provider } = (
+      integration as ThirdPartyAuthCallbackIntegration
+    ).thirdPartyAuthParams();
+
+    if (!thirdPartyOAuthCode) {
+      return hardNavigate('/');
+    }
+
+    try {
+      const linkedAccount: LinkedAccountData =
+        await account.verifyAccountThirdParty(
+          thirdPartyOAuthCode,
+          provider,
+          undefined,
+          queryParamsToMetricsContext(
+            flowQueryParams as unknown as Record<string, string>
+          )
+        );
+      await storeLinkedAccountData(linkedAccount);
+
+      setCurrentAccount(linkedAccount.uid);
+
+      const fxaParams = (
+        integration as ThirdPartyAuthCallbackIntegration
+      ).getFxAParams();
+
+      if (fxaParams.includes('client_id')) {
+        // HACK: Force the query params to be set in the URL, which then loads
+        // the integration stored in ThirdPartyAuthCallbackIntegration `state` param.
+        const urlQueryData = new UrlQueryData(new ReachRouterWindow());
+        urlQueryData.setParams(new URLSearchParams(fxaParams));
+      } else {
+        // If this isn't an OAuth flow, we can complete the login and navigate to settings
+        return performNavigation(linkedAccount);
+      }
+    } catch (error) {
+      // TODO validate what should happen here
+      hardNavigate('/');
```

Comment:
Looks like this matches what we have in backbone at the moment, are you thinking we should do something different?